threestudio/systems/fantasia3d.py

```diff
@@ -68,9 +68,15 @@ def training_step(self, batch, batch_idx):
             )
         else:  # texture training
             guidance_inp = out["comp_rgb"]
-            guidance_out = self.guidance(
-                guidance_inp, prompt_utils, **batch, rgb_as_latents=False
-            )
+            cond_inp = out["comp_normal"]
```

Comment:
How about moving this line under the `if`?

Reply:
Done!